src/routes/blog/post/integrate-sql-nosql-vector-graph-or-any-database-into-your-appwrite-project/+page.markdoc

```diff
@@ -0,0 +1,340 @@
+  const capacity = 10 + Math.round(Math.random() * 10) * 10; // Random number: 10,20,30,...,90,100
+
+  await client.db("main").collection("warehouses").insertOne({
+      location,
+      capacity
+  });
+
+  // Query data
+  const page = 1;
+  const limit = 100;
+  const cursor = client.db("main").collection("warehouses")
+    .find().limit(limit).skip((page - 1) * limit);
+
+  const docs = [];
+  for await (const doc of cursor) {
+    docs.push(doc);
+  }
+
+  return res.json(docs);
+};
+```
+
+A better-structured example can be found in the [query Mongo Atlas](https://github.com/appwrite/templates/tree/main/node/query-mongo-atlas) function template.
+
+![MongoDB Atlas Console](/images/blog/integrate-sql-nosql-vector-graph-or-any-database-into-your-appwrite-project/mongodb.png)
+
+## Redis Labs
+
+Redis is a widely used key-value database, most often serving as a cache layer in front of another database. It’s faster than the database behind it since Redis stores data in memory.
+
+While the cache is a typical use case for Redis, it can be used for much more since it can also store data persistently to drive. Its key-value approach gives very fast read and write operations, with the latest Redis version also supporting query abilities. Other use cases for Redis include pub/sub, atomic counters, distributed locks, or message queues.
+
+You can start integrating with Redis Labs by creating and connecting a client.
+
+```jsx
+import { createClient } from 'redis';
+import { createHash } from 'node:crypto';
+
+let client;
+
+export default async ({ req, res, log, error }) => {
+  // Connect to database
+  if(!client) {
+    const { REDIS_PASSWORD, REDIS_HOST } = process.env;
+    client = createClient({
+      password: REDIS_PASSWORD,
+      socket: {
+        host: REDIS_HOST,
+        port: 14714
+      }
+    });
+ 
+    await client.connect();
+  }
+
+  // Rest of the code
+};
+```
+
+Next, you can start inserting data. Since Redis is a key-value store, we need to create a unique key for every value. In the example below, an MD5 hash is used because the MD5 hash of the query is what’s commonly used as a key.
+
+```jsx
+// Insert data
+const location = `Street ${Math.round(Math.random() * 1000)}, Earth`; // Random address
+const capacity = 10 + Math.round(Math.random() * 10) * 10; // Random number: 10,20,30,...,90,100
+
+const hash = createHash('md5').update(location).digest('hex');
+await client.hSet(`warehouses:${hash}`, {
+  location,
+  capacity
+});
+```
+
+Querying data is as simple as fetching it using the key (MD5 hash). Many more operations are supported, such as JSON or array manipulation, but for simplicity, you can read a value by a key.
+
+```jsx
+// Query data
+const warehouses = await client.hGetAll(`warehouses:${hash}`);
+return res.json(warehouses);
+```
+
+A working example can be found in our [query Redis Labs](https://github.com/appwrite/templates/tree/main/node/query-redis-labs) function template.
```

Comment:
Any way to mention all these examples are also available through the function templates on the console? Because I left the same comment above for the other 2 dbs